File: MPP/ClientServer_Java/Services/src/main/java/services/IServer.java
package services;

import model.Categorie;
import model.Participant;
import model.Proba;
import model.User;
import model.dto.ProbaDTO;

import java.util.List;


public interface IServer {
    public void inscriereParticipant(String nume, int varsta, List<Proba> listaProbe, String usernameOperator) throws MyAppException;

    public Iterable<Participant> filtreazaParticipantiKeyword(String proba, String categorie) throws MyAppException;

    public Iterable<Proba> listaProbe() throws MyAppException;

    public ProbaDTO [] listaProbeDTO() throws MyAppException;

    public Iterable<Participant> listaParticipanti() throws MyAppException;

    public Iterable<Categorie> listaCategorii() throws MyAppException;

    public Iterable<String> listaProbeNume() throws MyAppException;

    public int nrParticipantiProba(Proba proba) throws MyAppException;

//    public int nrParticipantiCategorie(String categorie);

//    public boolean verificaCtg(int varsta, Proba proba);

    public User cauta(String username) throws MyAppException;

    public void login(String username, String password, IObserver client) throws MyAppException;

    public void logout(User user) throws MyAppException;

    //public User createUser(String username, String parola) throws PasswordStorage.CannotPerformOperationException ;

    //public User createUser() throws PasswordStorage.CannotPerformOperationException ;

}

File: MPP/ClientServer_C#/Service/IServer.cs
using Concurs.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
namespace Services
{
    public interface IServer
    {
        void Login(String username, String password, IObserver client);
        User Cauta(String username);
        void Logout(User user, IObserver client);
        IEnumerable<Participant> ListaParticipanti();
        IEnumerable<String> ListaProbeNume();
        IEnumerable<String> ListaCategorii();
        IEnumerable<Participant> FiltreazaParticipantiKeyword(string proba, string categorie);
        IEnumerable<Proba> ListaProbe();
        void InscriereParticipant(string nume, int varsta, List<Proba> listaProbe, string usernameOperator);
    }
}
